src/backend/marsha/core/api.py

```diff
@@ -69,8 +69,10 @@ def update_state(request):
     model = apps.get_model(app_label="core", model_name=key_elements["model_name"])
 
     extra_parameters = serializer.validated_data["extraParameters"]
-    if serializer.validated_data["state"] == defaults.READY and hasattr(
-        model, "extension"
+    if (
+        serializer.validated_data["state"] == defaults.READY
+        and hasattr(model, "extension")
+        and "extension" not in extra_parameters
```

Comment:
I don't understand this change and why the extension can be in key_elements :shrug:

Reply:
This was done for documents. And the extension is in the regex finding key elements allowing us to update the extension only when the document is ready.